plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/labeling/helper/StyledTextCalculationHelper.xtend

```diff
@@ -96,45 +95,54 @@ class StyledTextCalculationHelper {
 	def dispatch StyledString dispatchGetStyledText(EObjectWithContext objectWithContext) {
 		var StyledString styledText = dispatchGetStyledText(objectWithContext.obj);
 
-		if (objectWithContext.obj instanceof N4MemberDeclaration) {
-			val TMember member = (objectWithContext.obj as N4MemberDeclaration).definedTypeElement;
-			if (member !== null && member.containingType !== null &&
-				member.containingType != objectWithContext.context) {
+		val TMember member = if (objectWithContext.obj instanceof N4MemberDeclaration) {
+				(objectWithContext.obj as N4MemberDeclaration).definedTypeElement
+			} else if (objectWithContext.obj instanceof TMember) {
+				objectWithContext.obj as TMember
+			} else
+				null;
 
-				val orgDest = " from " + labelCalculationHelper.dispatchDoGetText(member.containingType);
+		if (member !== null && member.containingType !== null && member.containingType != objectWithContext.context) {
 
-				if (member.polyfilled) {
-					styledText.append(" polyfilled" + orgDest, N4JSStylers.POLYFILLED_MEMBERS_STYLER);
-				} else if (member.containingType instanceof TInterface && objectWithContext.context instanceof TClass) {
-					styledText.append(" consumed" + orgDest, N4JSStylers.CONSUMED_MEMBERS_STYLER);
-				} else {
-					styledText.append(" inherited" + orgDest, N4JSStylers.INHERITED_MEMBERS_STYLER);
-				}
+			val orgDest = " from " + labelCalculationHelper.dispatchDoGetText(member.containingType);
 
+			if (member.polyfilled) {
+				styledText.append(" polyfilled" + orgDest, N4JSStylers.POLYFILLED_MEMBERS_STYLER);
+			} else if (member.containingType instanceof TInterface && objectWithContext.context instanceof TClass) {
```

Comment:
If the member is not consumed, it does not show up in the list of members returned by the MemberCollector. So it only shows up if it actually is consumed. The only trick is to identify that it is consumed by a class, that means, the context (the TClass) must be different from the classifier (the TInterface) defining the member. If the member is abstract, we may get an error in the class (if it is not abstract), but we still want to see al members a client can access.